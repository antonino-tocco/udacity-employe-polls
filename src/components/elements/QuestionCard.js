import {Box, Button, Card, CardActions, CardContent, CardHeader} from "@mui/material";
import {useNavigate} from 'react-router';
import {DateTime} from 'luxon';

const QuestionCard = ({question, navigate}) => {


    const goToQuestionDetail = () => {
        console.log(`Go to question detail`);
        navigate(`/questions/${question.id}`);
    }


    return (<Card>
        <CardHeader title={question?.author} />
        <CardContent>
            <span>{DateTime.fromMillis(question?.timestamp).toFormat('dd-MM-yyyy | HH:mm:ss')}</span><br />
        </CardContent>
        <CardActions>
            <Box sx={{display: 'flex', flexGrow: 1, flexDirection: 'horizontal', justifyContent: 'center'}}>
                <Button sx={{width: '100%'}} variant='outlined' onClick={goToQuestionDetail}>SHOW</Button>
            </Box>
        </CardActions>
    </Card>);
}


export default QuestionCard;