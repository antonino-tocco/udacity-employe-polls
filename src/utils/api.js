import {_getQuestions, _getUsers, _saveQuestion} from "./_DATA";
import _get from "@babel/runtime/helpers/esm/get";

export async function retrieveUsers() {
    try {
        return await _getUsers();
    } catch (exception) {
        throw exception;
    }
}

export async function retrieveQuestions() {
    try {
        return await _getQuestions();
    } catch (exception) {
        throw exception;
    }
}

export async function retrieveQuestion(id) {
    try {
        const questions = await _getQuestions();
        if (!!questions && !!questions[id]) {
            return questions[id];
        }
        return null;
    } catch (exception) {
        throw exception;
    }
}

export async function saveQuestion(question) {
    try {
        await _saveQuestion(question);
    } catch (exception) {
        throw exception;
    }
}