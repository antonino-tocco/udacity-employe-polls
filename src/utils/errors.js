const errors = {
    USER_NOT_FOUND: 'USER_NOT_FOUND',
    PASSWORD_NOT_MATCH: 'PASSWORD_NOT_MATCH',
    QUESTION_NOT_FOUND: 'QUESTION_NOT_FOUND',
    QUESTION_BAD_FORMATTED: 'QUESTION_BAD_FORMATTED',
    ANSWER_BAD_FORMATTED: 'ANSWER_BAD_FORMATTED'
}

export default errors;