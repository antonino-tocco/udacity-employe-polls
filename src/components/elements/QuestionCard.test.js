import * as React from 'react';
import { render } from '@testing-library/react';
import QuestionCard from "./QuestionCard";
import * as api from '../../utils/api';


describe('QuestionCard component', () => {
    it('render QuestionCard with question ', async () => {
        const question = await api.retrieveQuestion('8xf0y6ziyjabvozdd253nd');
        const view = render(<QuestionCard question={question}/>);
        expect(view).toMatchSnapshot();
    })
})