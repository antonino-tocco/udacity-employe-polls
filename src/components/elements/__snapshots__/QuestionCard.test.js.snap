// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard component render QuestionCard with question  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              sarahedo
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <span>
            29-06-2016 | 04:21:12
          </span>
          <br />
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <div
            class="MuiBox-root css-115x9ip"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-xsqe0f-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              SHOW
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
          >
            sarahedo
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <span>
          29-06-2016 | 04:21:12
        </span>
        <br />
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
      >
        <div
          class="MuiBox-root css-115x9ip"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-xsqe0f-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            SHOW
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
